File: jdr-final/src/main/java/com/monapp/entity/Partie.java
package com.monapp.entity;


import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Partie {

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@JsonView(Views.Common.class)
	private int id;
	
	@ManyToOne(fetch=FetchType.EAGER)
	@JoinColumn(name="plateau")
	@JsonView(Views.Partie.class)
	private Plateau plateau;
	
	@OneToOne(fetch=FetchType.EAGER)
	@JoinColumn(name="personnage")
	@JsonView(Views.Partie.class)
	private Personnage personnage;
	

	
	@ManyToOne (fetch=FetchType.EAGER)
	@JoinColumn(name= "utilisateur")
	@JsonView(Views.Common.class)
	private Utilisateur utilisateur; 
	
	public Partie() {}
	public Partie(Plateau plateau, Personnage personnage) {
		this.plateau = plateau;
		this.personnage = personnage;
	}
	

	
	
}

File: app/partie.ts
import {Plateau} from './plateau';
import {Personnage} from './personnage';
import {Utilisateur} from './utilisateur';

export class Partie {

  constructor(  private _id: number,
                private _plateau: Plateau,
                private _personnage: Personnage,
                private _positionDepart: number,
                private _positionArrivee: number,
                private _utilisateur: Utilisateur) {}


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get plateau(): Plateau {
    return this._plateau;
  }

  set plateau(value: Plateau) {
    this._plateau = value;
  }

  get personnage(): Personnage {
    return this._personnage;
  }

  set personnage(value: Personnage) {
    this._personnage = value;
  }

  get positionDepart(): number {
    return this._positionDepart;
  }

  set positionDepart(value: number) {
    this._positionDepart = value;
  }

  get positionArrivee(): number {
    return this._positionArrivee;
  }

  set positionArrivee(value: number) {
    this._positionArrivee = value;
  }

  get utilisateur(): Utilisateur {
    return this._utilisateur;
  }

  set utilisateur(value: Utilisateur) {
    this._utilisateur = value;
  }
}
